core/src/main/java/org/apache/carbondata/core/datastore/chunk/impl/DimensionRawColumnChunk.java

```diff
@@ -168,13 +169,16 @@ public CarbonDictionary getLocalDictionary() {
    */
   public static CarbonDictionary getDictionary(LocalDictionaryChunk localDictionaryChunk,
       Compressor compressor) throws IOException, MemoryException {
+    boolean isUnsafe = Boolean.parseBoolean(CarbonProperties.getInstance()
```

Comment:
similar to other places below this is also a query execution flow...so here also the parameter `CarbonCommonConstants.ENABLE_UNSAFE_IN_QUERY_EXECUTION` need to be considered for deciding on unsafe or only one parameter is sufficient?